src/lib.rs

```diff
@@ -165,61 +166,113 @@ fn tornado_event_mapper(
     Ok(Some(tornado_events))
 }
 
+
+fn string_to_bigdecimal(amount: &str) -> Option<BigDecimal> {
```

Comment:
This is an overly complicated way of doing this, there are already helpers. `BigDecimal::from_str`